librecad/src/lib/math/lc_linemath.cpp

```diff
@@ -355,28 +339,9 @@ RS_LineData LC_LineMath::createParallel(RS_Vector &start, RS_Vector &end, double
  * @param e2  end of line 2
  * @return intersection point, invalid vector if no intersection point found
  */
-RS_Vector LC_LineMath::getIntersectionLineLine(RS_Vector& s1, RS_Vector& e1, RS_Vector& s2, RS_Vector& e2) {
-
-    RS_Vector ret;
-
-    double num = ((e2.x - s2.x) * (s1.y - s2.y) - (e2.y - s2.y) * (s1.x - s2.x));
-    double div = ((e2.y - s2.y) * (e1.x - s1.x) - (e2.x - s2.x) * (e1.y - s1.y));
-
-    double angle1 = s1.angleTo(e1);
-    double angle2 = s2.angleTo(e2);
-
-    if (fabs(div)>RS_TOLERANCE &&
-        fabs(remainder(angle1-angle2, M_PI))>=RS_TOLERANCE*10.) {
-        double u = num / div;
-
-        double xs = s1.x + u * (e1.x - s1.x);
-        double ys = s1.y + u * (e1.y - s1.y);
-        ret = RS_Vector(xs, ys);
-    }
-    else {
-        // lines are parallel
-        ret = RS_Vector(false);
-    }
-
-    return ret;
+RS_Vector LC_LineMath::getIntersectionLineLine(const RS_Vector& s1, const RS_Vector& e1, const RS_Vector& s2, const RS_Vector& e2) {
```

Comment:
yes, you're correct regarding the code duplication there. Yet  the reason why there is separate implementation is just a minor performance optimization - the original implementation does not create additional temporary objects and there might be only one intersection point (so one solution). So it just may be slightly faster. 
For sure, these implementations are equivalent, but I think it might be practical to add that specific case for lines intersections based on points to RS_Information class?